### ISSUE TITLE
Detect computed order declaration

### ISSUE THREAD

**Issue by thelinuxlich** [pmndrs/eslint-plugin-valtio#3]
If the user creates a computed field deriving value from another computed, the computed source should be declared first:

https://github.com/pmndrs/valtio/pull/84

**Comment by dai-shi** [comment]
That's a neat idea!!! Love it!

@thelinuxlich Would you like to work on it? (Or, leave it to someone??)

(Just noticed that @aslemammad is not watching this repo.)

**Comment by Aslemammad** [comment]
@dai-shi @thelinuxlich Now I found out that it's not on the watching list, sorry. That's great, I'm waiting for @thelinuxlich response, if he won't work on it, I'm here.

**Comment by thelinuxlich** [comment]
I wish I could, I'm drowning in work right now :(

**Comment by Aslemammad** [comment]
@dai-shi Could you give me an example of how it works?

**Comment by dai-shi** [comment]
@Aslemammad Wait a moment. We may introduce a breaking change. https://github.com/pmndrs/valtio/issues/86

**Comment by dai-shi** [comment]
Okay, I don't think we drop proxyWithComputed in any case.

So, what you would like to look at is:
- https://github.com/pmndrs/valtio/pull/84/files
- https://github.com/pmndrs/valtio/issues/82#issuecomment-778020202

**Comment by Aslemammad** [comment]
So it should warn:
```ts
}, {
  quadrupled: (snap) => snap.doubled * 2,
  doubled: (snap) => snap.count * 2,
// We have to put the doubled upon the quadrupled
})
```
Like that?

**Comment by dai-shi** [comment]
yes, that's what i understand.
